src/containers/settings/data/processing.js

```diff
@@ -0,0 +1,222 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2016, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+import React from 'react';
+
+import * as utilities from './utilities';
+
+export const settingsShape = React.PropTypes.shape({
+  activeSchedule: React.PropTypes.string.isRequired,
+  units: React.PropTypes.object.isRequired,
+  deviceId: React.PropTypes.string.isRequired,
+  basalSchedules: React.PropTypes.arrayOf(
+    React.PropTypes.shape({
+      name: React.PropTypes.string.isRequired,
+      value: React.PropTypes.arrayOf(
+        React.PropTypes.shape({
+          start: React.PropTypes.number.isRequired,
+          rate: React.PropTypes.number.isRequired,
+        }),
+      ),
+    }),
+  ).isRequired,
+  bgTarget: React.PropTypes.arrayOf(
+    React.PropTypes.shape({
+      start: React.PropTypes.number.isRequired,
+      target: React.PropTypes.number,
+      range: React.PropTypes.number,
+    })
+  ).isRequired,
+  carbRatio: React.PropTypes.arrayOf(
+    React.PropTypes.shape({
+      start: React.PropTypes.number.isRequired,
+      amount: React.PropTypes.number.isRequired,
+    })
+  ).isRequired,
+  insulinSensitivity: React.PropTypes.arrayOf(
+    React.PropTypes.shape({
+      start: React.PropTypes.number.isRequired,
+      amount: React.PropTypes.number.isRequired,
+    })
+  ).isRequired,
+});
+
+export const timedSettingsShape = React.PropTypes.shape({
+  activeSchedule: React.PropTypes.string.isRequired,
+  units: React.PropTypes.object.isRequired,
+  deviceId: React.PropTypes.string.isRequired,
+  basalSchedules: React.PropTypes.arrayOf(
+    React.PropTypes.shape({
+      name: React.PropTypes.string.isRequired,
+      value: React.PropTypes.arrayOf(
+        React.PropTypes.shape({
+          start: React.PropTypes.number.isRequired,
+          rate: React.PropTypes.number.isRequired,
+        }),
+      ),
+    }),
```

Comment:
I think you need to add an `isRequired` on this line to make sure the array contains at least one object with the given shape.